README.md

```diff
@@ -1,59 +1,106 @@
 
-## Integrate into Oniro-OpenHarmony build system
-- First of all, obtain the source code following [Oniro docs](https://docs.oniroproject.org/quick-build.html#obtaining-the-source-code).
-- Secondly, clone this repository into `//third_party/kanto`
+This project outlines the process for integrating third-party services, specifically Kanto and Mosquitto, into the Oniro-OpenHarmony build system. By following these instructions, developers can include these components into their Oniro-OpenHarmony environment, enhancing the system's capabilities with Kanto's suite connector and Mosquitto's message broker functionalities.
+
+## Prerequisites
+Before proceeding, ensure you have obtained the Oniro-OpenHarmony source code as per the [Oniro documentation](https://docs.oniroproject.org/quick-build.html#obtaining-the-source-code).
+
+## Integration Steps
+### Cloning Repositories
+Clone the Kanto and Mosquitto repositories into the `//third_party` directory to incorporate them into the Oniro-OpenHarmony system:
 
 ```console
 $ git clone --recurse-submodules  https://github.com/frankplus/third_party_kanto.git ./third_party/kanto
+$ git clone --recurse-submodules  https://github.com/frankplus/third_party_mosquitto.git ./third_party/mosquitto
 ```
 
-- add kanto component into the thirdparty subsystem by modifying `//productdefine/common/inherit/rich.json`
+### Modifying Configuration Files
+Add Kanto and Mosquitto components to the "thirdparty" subsystem by editing the `//productdefine/common/inherit/rich.json` file. This involves appending their configurations to the subsystem component list.
 
 ```diff
 diff --git a/inherit/rich.json b/inherit/rich.json
-index 255af7f..0075201 100644
+index 255af7f..e8ff335 100644
 --- a/inherit/rich.json
 +++ b/inherit/rich.json
-@@ -947,12 +947,16 @@
-     {
-       "subsystem": "thirdparty",
-       "components": [
+@@ -950,6 +950,14 @@
          {
            "component": "musl",
            "features": []
 +        },
 +        {
 +          "component": "kanto",
++          "features": []
++        },
++        {
++          "component": "mosquitto",
 +          "features": []
          }
        ]
      },
-     {
-       "subsystem": "hdf",
-       "components": [
+
 ```
 
-- build a new Docker image from the Dockerfile which, starting from OpenHarmony docker image, is adding golang. Go is required to build kanto suite connector.
+### Adjusting Security Settings
+Currently, Oniro-OpenHarmony does not natively support third-party services such as Kanto and Mosquitto. As a temporary measure, disable certain security features by modifying the `//vendor/hihope/rk3568/config.json` file.
+
+```diff
+diff --git a/rk3568/config.json b/rk3568/config.json
+index ea2d588..b6a8dc0 100755
+--- a/rk3568/config.json
++++ b/rk3568/config.json
+@@ -9,7 +9,7 @@
+   "api_version": 8,
+   "enable_ramdisk": true,
+   "enable_absystem": false,
+-  "build_selinux": true,
++  "build_selinux": false,
```

Comment:
Do we still need to disable this with the  "secon" : "u:r:kanto:s0" change in the next commit?

Reply:
updated instructions